engine/gamesys/src/gamesys/scripts/script_label.cpp

```diff
@@ -210,19 +211,23 @@ static int SetText(lua_State* L)
 
     dmGameObject::HInstance instance = CheckGoInstance(L);
 
-    size_t len = 0;
-    const char* text = luaL_checklstring(L, 2, &len);
+    size_t text_len = 0;
+    const char* text = luaL_checklstring(L, 2, &text_len);
     if (!text)
     {
         return DM_LUA_ERROR("Expected string as second argument");
     }
 
-    lua_newtable(L);
-    lua_pushlstring(L, text, len);
-    lua_setfield(L, -2, "text");
+    uint32_t data_size = sizeof(dmGameSystemDDF::SetText) + text_len + 1;
+    if (data_size > dmMessage::DM_MESSAGE_MAX_DATA_SIZE)
+    {
+        return DM_LUA_ERROR("The label string is too long!");
+    }
+    uint8_t data[data_size];
```

Comment:
However, they've also been moved to "optional" since C11 ([more info about VLA's](https://www.scaler.com/topics/c/dynamic-array-in-c/)).
And as such, I'd rather not use them.
Either use alloca(), of malloc a temporary object.

Reply:
I would accept a `data[dmMessage::DM_MESSAGE_MAX_DATA_SIZE]` too.

Reply:
@JCash Ok! I am not an expert in C++ compilers by any means but I would think that alloca() is preferable to static allocation because it generally uses much less RAM and maybe even just gets into cache entirely if it's small enough. With short texts for labels I think it is usually the case. But I understand it takes some CPU time for the allocation?
So which is preferable from the engine design view point?